Signup_Login_Assignment/src/main/java/com/example/signup_login_assignment/service/UserService.java

```diff
@@ -0,0 +1,107 @@
+package com.example.signup_login_assignment.service;
+
+import com.example.signup_login_assignment.dto.UserDto;
+import com.example.signup_login_assignment.entity.User;
+import com.example.signup_login_assignment.repository.UserRepository;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.regex.Pattern;
+
+
+@Service
+public class UserService {
+    private final UserRepository userRepository;
+    public UserService(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    String message = null;
+    BCryptPasswordEncoder b = new BCryptPasswordEncoder();
+
+    public String userRegistration(UserDto userDto) {
+        String phone = null, email = null;
+        if(userDto !=null) {
+            //
+            if (userRepository.existEmail(userDto.getEmail()) == null) {
+
+                if (!emailValidation(userDto.getEmail())) {
+                email = "Error";
+            }
+            if (!phoneValidation(userDto.getPhoneNo())) {
```

Comment:
if user give null phone no then....first check it is empty &  assign "Error" string in 'phone' variable(first IF) and then in second IF it will replace "Error" with "please enter valid mobile no " string